### ISSUE TITLE
ArduinoNano

### ISSUE THREAD

**Issue by EugenStery** [arthus701/midibigbutton#1]
Code can´t work on Arduino Nano, because of absence native USB HID capability?

**Comment by arthus701** [comment]
Yeah I don't know why it doesn't work with a nano, but you may be right. I am using an arduino micro and didn't get my hands on a nano yet, and probably will not in the near future. I think it's possible to modify the code to use the serial output and solder a MIDI jack to the arduino, so it can utilize "real" midi to do the same as it does now, but maybe you will run out of pins if you translate the sketch 1 to 1...I am gonna add a note to the readme to point out this issue...

**Comment by ghost** [comment]
Hi, I have a question. Does this project requires original arduino micro or some clone? I'm going to build this project, but on the internet the orignal arduino have 34 pins wheras the "copy" has only 24 pins. The board is Arduino Micro with ATmega32U4. Which one is right?

**Comment by arthus701** [comment]
Hi!
I used a noname clone of this
https://www.sparkfun.com/products/12640
ProMicro I bought on eBay. Please let me know how the project turns out for you :)

**Comment by ghost** [comment]
Thank you very much! Fortunately I have an opportunity to buy an exactly same one as u mentioned above. Of course I let you know how this tourns out. If I'm understand the whole concept correctly this modification should allow me to send midi clock from my synth to the Big Button, right?

**Comment by arthus701** [comment]
I never tested it with "actual" MIDI (i.e. with a MIDI cable), all I did was send MIDI via USB from ableton. I think it is possible to solder a MIDI jack to the serial pins of the Arduino, but you should do some research to validate this. I also think it's be possible to use a MIDI-USB adapter and send the MIDI clock from your synth to the Button this way. But as I said, I only used the Big Button for triggering drumracks in ableton...

**Comment by ghost** [comment]
Ok, yeah, I also think it should be fine. Now I am waiting for the ordered parts so when I finally get them and build everything I am gonna share results here. This whole concept is really interesting and I think it needs be more popular. For someone who want use Big Button little differently this thread should be really helpful.

**Comment by ripvanwinkle1** [comment]
Hi again, so after a long time i finally get the board, but when i tried upload .ino file via usb i get an error:

Arduino:1.8.14 Hourly Build 2020/06/24 01:33 (Windows 7), Board:"Arduino Micro"

MIDI_BIG_BUTTON:40:10: fatal error: MIDIUSB.h: No such file or directory

 #include "MIDIUSB.h"

          ^~~~~~~~~~~

compilation terminated.

exit status 1

MIDIUSB.h: No such file or directory

Really, I am kinda dumb when it comes to arduino programming so maybe you know what kinda problem is this? When I verified code the program also showed similar error.

**Comment by arthus701** [comment]
So I cannot see your comment anymore, but I got an email stating that you have trouble using usbmidi. From your error log it appears you do not have the library installed. In the arduino editor go to Tools->Manage Libraries and type "usbmidi", then install and you should be good to go :)

**Comment by ghost** [comment]
Yeah, I deleted my comment becaus I figured out that I forgot to put usbmidi library. All instalization process went without any troubles so I think it's okay. Buttons and jacks seems working. Now I'm goning to test this thing like you did ( connect to some kind of sowtware on computer). Then if everything is ok, I'll attempt to connect sequencer nad my master device. (i also working on usb midi converter thing in order to make connection)

**Comment by ghost** [comment]
Hi, I have a couple questions. Did you power the sequencer via usb or like lmnc did by power socket? And what did you using for trigger device? Currently my sequencer is working but the clock timing is really weird - i think it is because I somethin screw up in abelton or more probably in my trigger device.

**Comment by arthus701** [comment]
Hi!
I did power it via USB, as I only used it with ableton. But a quick google search suggested it's not hard to power it via [battery](https://forum.arduino.cc/index.php?topic=209453.0) I triggered it using ableton's sync out. As you can see from [this line of code](https://github.com/arthus701/midibigbutton/blob/650756b4473f9ce92bfa2ee645c20158b36a984d/MIDI_BIG_BUTTON.ino#L196), a quarter note consists of 24 pulses.  
I own a Minibrute 2S (which I never used with the Big Button), which has several clock modes and one is 24ppq (which I assume means Pulses Per Quarter), so I think this is some kind of standard. Maybe you can find similar settings for your sequencer? Good luck and thanks for building this thing, it means a lot to me :)

**Comment by ghost** [comment]
Ok, and is it possbile to add clock input like it was in original? Because what i see is that this input is replaced by midiusb clock input? Correct me if I'm wrong. And also the "old" shift knob is replaced by speed knob when it comes to its function, right?

**Comment by arthus701** [comment]
Adding a 2nd clock in is a bit tricky, but I will look into it tomorrow.
Yeah, I changed the functionality of the shift knob, you’re right!

**Comment by ghost** [comment]
ok, right now I'm fine with only midiclock option - well, that is the main reason why I build this. This is my first sequencer ever like really ever (don't wanna spend money haha) and I want to know about this thing as much as possible.

**Comment by arthus701** [comment]
So I managed to implement an analog clock-jack additionally to the usbmidi connection. However there are some issues with it. Maybe once you actually need it we can have a chat on some other medium and discuss your requirements. If this suits you, just comment once the time comes and maybe leave an email address or something.

**Comment by ghost** [comment]
Thank you, this might be useful for me but right now I'm fine. One thing very important for me. How you manage to triggering track from ableton? Like, big button has 6 jack outs. Is usbmidi is also capable for receiving samples? I'm really confused.

**Comment by arthus701** [comment]
The code in the repo does not only send a pulse to the LED/jack, but also sends MIDI notes via USB. The root note [is 56](https://github.com/arthus701/midibigbutton/blob/650756b4473f9ce92bfa2ee645c20158b36a984d/MIDI_BIG_BUTTON.ino#L82), and the channels are defined [here](https://github.com/arthus701/midibigbutton/blob/650756b4473f9ce92bfa2ee645c20158b36a984d/MIDI_BIG_BUTTON.ino#L100). In ableton it's easy to see this, when you have a drumrack in a midi track and just look at the matrix. You have to activate the arduino for midi input in the settings first.

**Comment by ghost** [comment]
Aa, okay, so what can I say other that this is wonderful project. Everything works so right now I better learn to use Ableton. Thanks, I was not shure at first if this project works but right now I'm convinced that midibigbutton was the best idea!
